tests/unit/test_utils.py

```diff
@@ -409,3 +410,25 @@ def test_can_escape_recursive_shapes(self):
         }
         self.walker.walk(self.get_shape_model('Recursive'), self.visitor)
         self.assert_visited_shapes(['Recursive'])
+
+
+@pytest.fixture(scope='module')
+def argument_model():
```

Comment:
For pytest fixtures, let's keep it at the top of the file in between the imports and test cases. The main reason I'm suggesting it is that when the fixtures are mixed between test cases it can be hard to realize that there is a fixture available and we may want to use it in the other test cases in the future. So keeping the fixtures all near the top, makes it easier to find them.